[StructuralMechanics] problem with membrane

Some of the membrane tests are failing atm after #2101 

~~~
======================================================================
FAIL: test_execution (structural_mechanics_test_factory.Fofi4PointTentnoCableTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/philipp/software/Kratos_MasterBranch/applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py", line 58, in test_execution
    self.test.RunSolutionLoop()
  File "/home/philipp/software/Kratos_MasterBranch/kratos/python_scripts/analysis_stage.py", line 60, in RunSolutionLoop
    self.FinalizeSolutionStep()
  File "/home/philipp/software/Kratos_MasterBranch/kratos/python_scripts/analysis_stage.py", line 130, in FinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "/home/philipp/software/Kratos_MasterBranch/kratos/python_scripts/from_json_check_result_process.py", line 156, in ExecuteFinalizeSolutionStep
    self.assertTrue(isclosethis, msg=(str(value) + " != " + str(value_json) + ", rel_tol = " + str(reltol) + ", abs_tol = " + str(tol) + " : Error checking node " + str(node.Id) + " " + variable_name + " results."))
AssertionError: False is not true : 0.0 != 0.07986688763000538, rel_tol = 1e-06, abs_tol = 0.001 : Error checking node 4 DISPLACEMENT_Z results.

======================================================================
FAIL: test_execution (structural_mechanics_test_factory.Fofi4PointTentCableTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/philipp/software/Kratos_MasterBranch/applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py", line 58, in test_execution
    self.test.RunSolutionLoop()
  File "/home/philipp/software/Kratos_MasterBranch/kratos/python_scripts/analysis_stage.py", line 60, in RunSolutionLoop
    self.FinalizeSolutionStep()
  File "/home/philipp/software/Kratos_MasterBranch/kratos/python_scripts/analysis_stage.py", line 130, in FinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "/home/philipp/software/Kratos_MasterBranch/kratos/python_scripts/from_json_check_result_process.py", line 156, in ExecuteFinalizeSolutionStep
    self.assertTrue(isclosethis, msg=(str(value) + " != " + str(value_json) + ", rel_tol = " + str(reltol) + ", abs_tol = " + str(tol) + " : Error checking node " + str(node.Id) + " " + variable_name + " results."))
AssertionError: False is not true : 0.0 != -0.03131864022032899, rel_tol = 1e-06, abs_tol = 0.001 : Error checking node 2 DISPLACEMENT_Z results.

======================================================================
FAIL: test_execution (structural_mechanics_test_factory.MembraneQ4PointLoadTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/philipp/software/Kratos_MasterBranch/applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py", line 58, in test_execution
    self.test.RunSolutionLoop()
  File "/home/philipp/software/Kratos_MasterBranch/kratos/python_scripts/analysis_stage.py", line 60, in RunSolutionLoop
    self.FinalizeSolutionStep()
  File "/home/philipp/software/Kratos_MasterBranch/kratos/python_scripts/analysis_stage.py", line 130, in FinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "/home/philipp/software/Kratos_MasterBranch/kratos/python_scripts/from_json_check_result_process.py", line 156, in ExecuteFinalizeSolutionStep
    self.assertTrue(isclosethis, msg=(str(value) + " != " + str(value_json) + ", rel_tol = " + str(reltol) + ", abs_tol = " + str(tol) + " : Error checking node " + str(node.Id) + " " + variable_name + " results."))
AssertionError: False is not true : 0.0 != -0.473612074593017, rel_tol = 1e-06, abs_tol = 0.001 : Error checking node 4 DISPLACEMENT_Z results.

======================================================================
FAIL: test_execution (structural_mechanics_test_factory.MembraneQ4TrussPointLoadTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/philipp/software/Kratos_MasterBranch/applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py", line 58, in test_execution
    self.test.RunSolutionLoop()
  File "/home/philipp/software/Kratos_MasterBranch/kratos/python_scripts/analysis_stage.py", line 60, in RunSolutionLoop
    self.FinalizeSolutionStep()
  File "/home/philipp/software/Kratos_MasterBranch/kratos/python_scripts/analysis_stage.py", line 130, in FinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "/home/philipp/software/Kratos_MasterBranch/kratos/python_scripts/from_json_check_result_process.py", line 156, in ExecuteFinalizeSolutionStep
    self.assertTrue(isclosethis, msg=(str(value) + " != " + str(value_json) + ", rel_tol = " + str(reltol) + ", abs_tol = " + str(tol) + " : Error checking node " + str(node.Id) + " " + variable_name + " results."))
AssertionError: False is not true : 0.0 != -0.037499751604797626, rel_tol = 1e-06, abs_tol = 0.001 : Error checking node 4 DISPLACEMENT_Z results.
~~~

fyi @inigolcanalejo @longchentum

--- a/applications/StructuralMechanicsApplication/custom_elements/prestress_membrane_element.cpp
+++ b/applications/StructuralMechanicsApplication/custom_elements/prestress_membrane_element.cpp
@@ -444,7 +444,7 @@ void PrestressMembraneElement::CalculateAndAddNonlinearKm(
 
 void PrestressMembraneElement::CalculateQ(
     BoundedMatrix<double, 3, 3>& Q,
-    const unsigned int& rPointNumber)
+    const unsigned int rPointNumber)
 
 {
     KRATOS_TRY
@@ -589,8 +589,8 @@ void PrestressMembraneElement::CalculateAll(
     MatrixType& rLeftHandSideMatrix,
     VectorType& rRightHandSideVector,
     const ProcessInfo& rCurrentProcessInfo,
-    const bool& rCalculateStiffnessMatrixFlag,
-    const bool& rCalculateResidualVectorFlag)
+    const bool rCalculateStiffnessMatrixFlag,
+    const bool rCalculateResidualVectorFlag)
 
 {
     KRATOS_TRY
@@ -744,7 +744,7 @@ void PrestressMembraneElement::GetValueOnIntegrationPoints(const Variable<Matrix
 }
 //***********************************************************************************
 //***********************************************************************************
-void PrestressMembraneElement::CalculateMetricDeformed(const unsigned int& rPointNumber, Matrix DN_De,
+void PrestressMembraneElement::CalculateMetricDeformed(const unsigned int rPointNumber, Matrix DN_De,
     array_1d<double, 3>& rgab,
     array_1d<double, 3>& rg1,
     array_1d<double, 3>& rg2)
@@ -820,7 +820,7 @@ void PrestressMembraneElement::CalculateSecondVariationStrain(Matrix DN_De,
 //***********************************************************************************
 //***********************************************************************************
 void PrestressMembraneElement::ProjectPrestress(
-    const unsigned int& rPointNumber){
+    const unsigned int rPointNumber){
 
     // Compute local cartesian basis E1, E2, E3
     array_1d<double,3> E1 = column( GetValue(BASE_REF_1),rPointNumber )/norm_2(column( GetValue(BASE_REF_1),rPointNumber ));
@@ -834,7 +834,8 @@ void PrestressMembraneElement::ProjectPrestress(
     array_1d<double, 3> T1, T2, T3;
 
     // Alternative 1: planar projection according to dissertation Roland Wuechner
-    if(GetProperties()[PROJECTION_TYPE_COMBO] == "planar"){
+    const std::string this_proyection_type = GetProperties().Has(PROJECTION_TYPE_COMBO) ? GetProperties()[PROJECTION_TYPE_COMBO] : "planar";
+    if(this_proyection_type == "planar"){
         // definition prestress axes
         array_1d<double,3> global_prestress_axis1, global_prestress_axis2, global_prestress_axis3;
         for(unsigned int i=0; i<3;i++){
@@ -842,10 +843,8 @@ void PrestressMembraneElement::ProjectPrestress(
             global_prestress_axis2[i] = GetValue(PRESTRESS_AXIS_2)(i,rPointNumber);
         }
 
-        #ifdef KRATOS_DEBUG
-            KRATOS_ERROR_IF(norm_2(global_prestress_axis1) < std::numeric_limits<double>::epsilon() && norm_2(global_prestress_axis1) > -std::numeric_limits<double>::epsilon()) << "division by zero!" << std::endl;
-            KRATOS_ERROR_IF(norm_2(global_prestress_axis2) < std::numeric_limits<double>::epsilon() && norm_2(global_prestress_axis2) > -std::numeric_limits<double>::epsilon()) << "division by zero!" << std::endl;
-        #endif
+        KRATOS_DEBUG_ERROR_IF(norm_2(global_prestress_axis1) < std::numeric_limits<double>::epsilon() && norm_2(global_prestress_axis1) > -std::numeric_limits<double>::epsilon()) << "division by zero!" << std::endl;
+        KRATOS_DEBUG_ERROR_IF(norm_2(global_prestress_axis2) < std::numeric_limits<double>::epsilon() && norm_2(global_prestress_axis2) > -std::numeric_limits<double>::epsilon()) << "division by zero!" << std::endl;
 
         // normalization global prestress axes
         global_prestress_axis1 /= norm_2(global_prestress_axis1);
@@ -863,18 +862,13 @@ void PrestressMembraneElement::ProjectPrestress(
         MathUtils<double>::CrossProduct(T2,E3,T1);
         T2 /= norm_2(T2);
         T3 = E3;
-    }
-
-    // Alternative 2: radial projection (T1 = radial direction, T3 = out-of-plane direction, T2=T3xT1)
-    if(GetProperties()[PROJECTION_TYPE_COMBO] == "radial"){
+    } else if(this_proyection_type == "radial"){ // Alternative 2: radial projection (T1 = radial direction, T3 = out-of-plane direction, T2=T3xT1)
         // definition prestress axes
         array_1d<double,3> global_prestress_axis1; // = direction of rotational axis
         for(unsigned int i=0; i<3;i++)
             global_prestress_axis1[i] = GetValue(PRESTRESS_AXIS_1)(i,rPointNumber);
 
-        #ifdef KRATOS_DEBUG
-            KRATOS_ERROR_IF(norm_2(global_prestress_axis1) < std::numeric_limits<double>::epsilon() && norm_2(global_prestress_axis1) > -std::numeric_limits<double>::epsilon()) << "division by zero!" << std::endl;
-        #endif
+        KRATOS_DEBUG_ERROR_IF(norm_2(global_prestress_axis1) < std::numeric_limits<double>::epsilon() && norm_2(global_prestress_axis1) > -std::numeric_limits<double>::epsilon()) << "division by zero!" << std::endl;
 
         // compute T1, T2, T3
         MathUtils<double>::CrossProduct(T1, global_prestress_axis1, E3);
@@ -882,8 +876,6 @@ void PrestressMembraneElement::ProjectPrestress(
         MathUtils<double>::CrossProduct(T2, E3, T1);
         T2 /= norm_2(T2);
         T3 = E3;
-
-
     }
 
     // Transform prestresses in the local cartesian cosy in reference configuration
@@ -946,7 +938,7 @@ void PrestressMembraneElement::InitializeNonLinearIteration(ProcessInfo& rCurren
 //***********************************************************************************
 //***********************************************************************************
 
-void PrestressMembraneElement::InitializeMaterial(const unsigned int& NumberIntegrationPoints){
+void PrestressMembraneElement::InitializeMaterial(const unsigned int NumberIntegrationPoints){
     if (mConstitutiveLawVector.size() == 0)
     {
         mConstitutiveLawVector.resize(NumberIntegrationPoints);
@@ -962,7 +954,7 @@ void PrestressMembraneElement::InitializeMaterial(const unsigned int& NumberInte
 
 //***********************************************************************************
 //***********************************************************************************
-void PrestressMembraneElement::UpdatePrestress(const unsigned int& rPointNumber){
+void PrestressMembraneElement::UpdatePrestress(const unsigned int rPointNumber){
     // --1--computation relevant CoSys
     array_1d<double, 3> g1, g2, g3, gab; //base vectors/metric actual config
     array_1d<double, 3> G3;  // base vector reference config
@@ -1014,7 +1006,7 @@ void PrestressMembraneElement::UpdatePrestress(const unsigned int& rPointNumber)
 //***********************************************************************************
 //***********************************************************************************
 
-void PrestressMembraneElement::ComputeRelevantCoSys(const unsigned int& rPointNumber,
+void PrestressMembraneElement::ComputeRelevantCoSys(const unsigned int rPointNumber,
              array_1d<double, 3>& rg1,array_1d<double, 3>& rg2,array_1d<double, 3>& rg3, array_1d<double, 3>& rgab,
              array_1d<double, 3>& rG3,
              array_1d<double, 3>& rE1Tot, array_1d<double, 3>& rE2Tot,array_1d<double, 3>& rE3Tot,
@@ -1057,9 +1049,7 @@ void PrestressMembraneElement::ComputeRelevantCoSys(const unsigned int& rPointNu
     // -->Invert metric in (total) reference configuration
     double det_metric_tot = metric_reference_tot(0)*metric_reference_tot(1)-metric_reference_tot(2)*metric_reference_tot(2);
     array_1d<double, 3> inv_metric_tot;
-    #ifdef KRATOS_DEBUG
-        KRATOS_ERROR_IF(det_metric_tot < std::numeric_limits<double>::epsilon() && det_metric_tot > -std::numeric_limits<double>::epsilon()) << "division by zero!" << std::endl;
-    #endif
+    KRATOS_DEBUG_ERROR_IF(det_metric_tot < std::numeric_limits<double>::epsilon() && det_metric_tot > -std::numeric_limits<double>::epsilon()) << "division by zero!" << std::endl;
     inv_metric_tot(0) = 1.0/det_metric_tot * metric_reference_tot(1);
     inv_metric_tot(1) = 1.0/det_metric_tot * metric_reference_tot(0);
     inv_metric_tot(2) = -1.0/det_metric_tot * metric_reference_tot(2);
@@ -1072,7 +1062,7 @@ void PrestressMembraneElement::ComputeRelevantCoSys(const unsigned int& rPointNu
 //***********************************************************************************
 //***********************************************************************************
 
-void PrestressMembraneElement::ComputeEigenvaluesDeformationGradient(const unsigned int& rPointNumber,
+void PrestressMembraneElement::ComputeEigenvaluesDeformationGradient(const unsigned int rPointNumber,
                     BoundedMatrix<double,3,3>& rOrigin, BoundedMatrix<double,3,3>& rTarget, BoundedMatrix<double,3,3>& rTensor,
                     const array_1d<double, 3>& rBaseRefContraTot1, const array_1d<double, 3>& rBaseRefContraTot2,
                     const array_1d<double, 3>& rE1Tot, const array_1d<double, 3>& rE2Tot, const array_1d<double, 3>& rE3Tot,
@@ -1111,7 +1101,7 @@ void PrestressMembraneElement::ComputeEigenvaluesDeformationGradient(const unsig
 //***********************************************************************************
 //***********************************************************************************
 
-void PrestressMembraneElement::ComputeEigenvectorsDeformationGradient(const unsigned int& rPointNumber,
+void PrestressMembraneElement::ComputeEigenvectorsDeformationGradient(const unsigned int rPointNumber,
                                 BoundedMatrix<double,3,3>& rTensor, BoundedMatrix<double,3,3>& rOrigin,
                                 const BoundedMatrix<double,3,3>& rDeformationGradientTotal,
                                 const array_1d<double, 3>& rE1Tot, const array_1d<double, 3>& rE2Tot,
@@ -1194,7 +1184,7 @@ void PrestressMembraneElement::ComputeEigenvectorsDeformationGradient(const unsi
 //***********************************************************************************
 //***********************************************************************************
 
-void PrestressMembraneElement::ModifyPrestress(const unsigned int& rPointNumber,
+void PrestressMembraneElement::ModifyPrestress(const unsigned int rPointNumber,
                     BoundedMatrix<double,3,3>& rOrigin, BoundedMatrix<double,3,3>& rTarget,BoundedMatrix<double,3,3>& rTensor,
                     const array_1d<double, 3>& rE1, const array_1d<double, 3>& rE2, const array_1d<double, 3>& rE3, const array_1d<double, 3>& rG3,
                     const array_1d<double, 3>& rg1, const array_1d<double, 3>& rg2, const array_1d<double, 3>& rg3, const BoundedMatrix<double,3,3>& rNAct,
@@ -1284,7 +1274,7 @@ void PrestressMembraneElement::ModifyPrestress(const unsigned int& rPointNumber,
     prestress_modified(2,rPointNumber) = rTensor(1,0);
 
 }
-void PrestressMembraneElement::ComputePrestress(const unsigned int& rIntegrationPointSize){
+void PrestressMembraneElement::ComputePrestress(const unsigned int rIntegrationPointSize){
     // initialize prestress matrix and prestress directions (with dummy zero values)
     unsigned int strain_size = this->GetProperties().GetValue(CONSTITUTIVE_LAW)->GetStrainSize();
     Matrix prestress_matrix(strain_size,rIntegrationPointSize,0), prestress_direction1(3,rIntegrationPointSize,0), prestress_direction2(3,rIntegrationPointSize,0);
@@ -1293,53 +1283,52 @@ void PrestressMembraneElement::ComputePrestress(const unsigned int& rIntegration
     this->SetValue(PRESTRESS_AXIS_1, prestress_direction1);
     this->SetValue(PRESTRESS_AXIS_2, prestress_direction2);
 
-    if(GetProperties().Has(PROJECTION_TYPE_COMBO)){
-        if(GetProperties()[PROJECTION_TYPE_COMBO] != "file"){
-            if(GetProperties().Has(PRESTRESS_VECTOR)){
-                Matrix& prestress_variable = this->GetValue(MEMBRANE_PRESTRESS);
-                Matrix& prestress_axis_1 = this->GetValue(PRESTRESS_AXIS_1);
-                Matrix& prestress_axis_2 = this->GetValue(PRESTRESS_AXIS_2);
+    const bool to_compute = GetProperties().Has(PROJECTION_TYPE_COMBO) ? GetProperties()[PROJECTION_TYPE_COMBO] != "file": true;
 
-                if(std::abs(GetProperties()[PRESTRESS_VECTOR](0)- GetProperties()[PRESTRESS_VECTOR](1)) <std::numeric_limits<double>::epsilon() && GetProperties()[PRESTRESS_VECTOR](2) == 0)
-                    mAnisotropicPrestress = false;
+    if(to_compute) {
+        if(GetProperties().Has(PRESTRESS_VECTOR)){
+            Matrix& prestress_variable = this->GetValue(MEMBRANE_PRESTRESS);
+            Matrix& prestress_axis_1 = this->GetValue(PRESTRESS_AXIS_1);
+            Matrix& prestress_axis_2 = this->GetValue(PRESTRESS_AXIS_2);
 
-                else
-                    mAnisotropicPrestress = true;
+            if(std::abs(GetProperties()[PRESTRESS_VECTOR](0)- GetProperties()[PRESTRESS_VECTOR](1)) <std::numeric_limits<double>::epsilon() && GetProperties()[PRESTRESS_VECTOR](2) == 0)
+                mAnisotropicPrestress = false;
 
+            else
+                mAnisotropicPrestress = true;
 
-                // read prestress from the material properties
-                for(unsigned int point_number = 0; point_number < rIntegrationPointSize;point_number ++){
-                    // anisotropic case: prestress projection in the membrane
-                    if(mAnisotropicPrestress) {
-                        // Initialize Prestress
-                        for (unsigned int i_strain=0; i_strain<strain_size; i_strain++){
-                            prestress_variable(i_strain,point_number) = GetProperties()[PRESTRESS_VECTOR](i_strain);
-                            if(GetProperties().Has(PRESTRESS_AXIS_1_GLOBAL))
-                                prestress_axis_1(i_strain,point_number) = GetProperties()[PRESTRESS_AXIS_1_GLOBAL](i_strain);
-                            if(GetProperties().Has(PRESTRESS_AXIS_2_GLOBAL))
-                                prestress_axis_2(i_strain,point_number) = GetProperties()[PRESTRESS_AXIS_2_GLOBAL](i_strain);
-                        }
-                        // in case that no prestress directions are prescribed: hardcode the prestress direction
-                        if (GetProperties().Has(PRESTRESS_AXIS_1_GLOBAL) == false){
-                            prestress_axis_1(0,point_number) = 1;
-                            prestress_axis_1(1,point_number) = 0;
-                            prestress_axis_1(2,point_number) = 0;
-                        }
-                        if (GetProperties().Has(PRESTRESS_AXIS_2_GLOBAL) == false){
-                            prestress_axis_2(0,point_number) = 0;
-                            prestress_axis_2(1,point_number) = 1;
-                            prestress_axis_2(2,point_number) = 0;
-                        }
-
-                        // Project prestress in membrane plane
-                        ProjectPrestress(point_number);
+            // read prestress from the material properties
+            for(unsigned int point_number = 0; point_number < rIntegrationPointSize;point_number ++){
+                // anisotropic case: prestress projection in the membrane
+                if(mAnisotropicPrestress) {
+                    // Initialize Prestress
+                    for (unsigned int i_strain=0; i_strain<strain_size; i_strain++){
+                        prestress_variable(i_strain,point_number) = GetProperties()[PRESTRESS_VECTOR](i_strain);
+                        if(GetProperties().Has(PRESTRESS_AXIS_1_GLOBAL))
+                            prestress_axis_1(i_strain,point_number) = GetProperties()[PRESTRESS_AXIS_1_GLOBAL](i_strain);
+                        if(GetProperties().Has(PRESTRESS_AXIS_2_GLOBAL))
+                            prestress_axis_2(i_strain,point_number) = GetProperties()[PRESTRESS_AXIS_2_GLOBAL](i_strain);
+                    }
+                    // in case that no prestress directions are prescribed: hardcode the prestress direction
+                    if (GetProperties().Has(PRESTRESS_AXIS_1_GLOBAL) == false){
+                        prestress_axis_1(0,point_number) = 1;
+                        prestress_axis_1(1,point_number) = 0;
+                        prestress_axis_1(2,point_number) = 0;
+                    }
+                    if (GetProperties().Has(PRESTRESS_AXIS_2_GLOBAL) == false){
+                        prestress_axis_2(0,point_number) = 0;
+                        prestress_axis_2(1,point_number) = 1;
+                        prestress_axis_2(2,point_number) = 0;
                     }
 
-                    // in case of isotropic prestress: set prestress in the first step (no transformation necessary)
-                    else {
-                        for (unsigned int i_strain=0; i_strain<strain_size; i_strain++){
-                            prestress_variable(i_strain,point_number) = GetProperties()[PRESTRESS_VECTOR](i_strain);
-                        }
+                    // Project prestress in membrane plane
+                    ProjectPrestress(point_number);
+                }
+
+                // in case of isotropic prestress: set prestress in the first step (no transformation necessary)
+                else {
+                    for (unsigned int i_strain=0; i_strain<strain_size; i_strain++){
+                        prestress_variable(i_strain,point_number) = GetProperties()[PRESTRESS_VECTOR](i_strain);
                     }
                 }
             }
@@ -1433,7 +1422,7 @@ void PrestressMembraneElement::ComputeBaseVectors(const GeometryType::Integratio
 void PrestressMembraneElement::ComputeContravariantBaseVectors(
                         array_1d<double, 3>& rG1Contra,
                         array_1d<double, 3>& rG2Contra,
-                        const unsigned int& rPointNumber){
+                        const unsigned int rPointNumber){
     // determinant metric
     double det_metric = mGab0[rPointNumber][0]*mGab0[rPointNumber][1]- mGab0[rPointNumber][2]*mGab0[rPointNumber][2];
 
@@ -1450,7 +1439,7 @@ void PrestressMembraneElement::ComputeContravariantBaseVectors(
 //***********************************************************************************
 //***********************************************************************************
 
-const Matrix PrestressMembraneElement::CalculateDeformationGradient(const unsigned int& rPointNumber){
+const Matrix PrestressMembraneElement::CalculateDeformationGradient(const unsigned int rPointNumber){
     // Compute contravariant base vectors in reference configuration
     array_1d<double, 3> G1_contra, G2_contra;
     ComputeContravariantBaseVectors(G1_contra, G2_contra, rPointNumber);
@@ -1480,7 +1469,7 @@ const Matrix PrestressMembraneElement::CalculateDeformationGradient(const unsign
 //***********************************************************************************
 //***********************************************************************************
 
-void PrestressMembraneElement::InitializeFormfinding(const unsigned int& rIntegrationPointSize){
+void PrestressMembraneElement::InitializeFormfinding(const unsigned int rIntegrationPointSize){
 
     if(mAnisotropicPrestress == true){
         // store base vectors of the initial configuration

--- a/applications/StructuralMechanicsApplication/custom_elements/prestress_membrane_element.hpp
+++ b/applications/StructuralMechanicsApplication/custom_elements/prestress_membrane_element.hpp
@@ -149,8 +149,8 @@ namespace Kratos
       MatrixType& rLeftHandSideMatrix,
       VectorType& rRightHandSideVector,
       const ProcessInfo& rCurrentProcessInfo,
-      const bool& rCalculateStiffnessMatrixFlag,
-      const bool& rCalculateResidualVectorFlag);
+      const bool CalculateStiffnessMatrixFlag,
+      const bool CalculateResidualVectorFlag);
 
     void CalculateAndAddKm(
       Matrix& rK,
@@ -171,7 +171,7 @@ namespace Kratos
 
     void CalculateQ(
       BoundedMatrix<double, 3, 3>& rQ,
-      const unsigned int& rPointNumber);
+      const unsigned int PointNumber);
 
     void CalculateB(
         Matrix& B,
@@ -200,7 +200,7 @@ namespace Kratos
       const double& rWeight,
       const ProcessInfo& rCurrentProcessInfo);
 
-    void CalculateMetricDeformed(const unsigned int& rPointNumber,
+    void CalculateMetricDeformed(const unsigned int PointNumber,
         Matrix DN_De,
         array_1d<double, 3>& rgab,
         array_1d<double, 3>& rg1,
@@ -213,51 +213,51 @@ namespace Kratos
         Matrix& Strain_locCartesian12,
         BoundedMatrix<double, 3, 3>& Q);
 
-    void InitializeFormfinding(const unsigned int& rIntegrationPointSize);
+    void InitializeFormfinding(const unsigned int rIntegrationPointSize);
 
-    void ProjectPrestress(const unsigned int& rPointNumber);
+    void ProjectPrestress(const unsigned int PointNumber);
 
-    void UpdatePrestress(const unsigned int& rPointNumber);
+    void UpdatePrestress(const unsigned int PointNumber);
 
-    void ComputePrestress(const unsigned int& rIntegrationPointSize);
+    void ComputePrestress(const unsigned int rIntegrationPointSize);
 
     void ComputeBaseVectors(const GeometryType::IntegrationPointsArrayType& rIntegrationPoints);
 
-    void InitializeMaterial(const unsigned int& NumberIntegrationPoints);
+    void InitializeMaterial(const unsigned int NumberIntegrationPoints);
 
     void ComputeContravariantBaseVectors(
                         array_1d<double, 3>& rG1Contra,
                         array_1d<double, 3>& rG2Contra,
-                        const unsigned int& rPointNumber);
+                        const unsigned int PointNumber);
 
-    void ComputeRelevantCoSys(const unsigned int& rPointNumber,
+    void ComputeRelevantCoSys(const unsigned int PointNumber,
              array_1d<double, 3>& rg1,array_1d<double, 3>& rg2,array_1d<double, 3>& rg3, array_1d<double, 3>& rgab,
              array_1d<double, 3>& rG3,
              array_1d<double, 3>& rE1Tot, array_1d<double, 3>& rE2Tot,array_1d<double, 3>& rE3Tot,
              array_1d<double, 3>& rE1,array_1d<double, 3>& rE2,array_1d<double, 3>& rE3,
              array_1d<double, 3>& rBaseRefContraTot1,array_1d<double, 3>& rBaseRefContraTot2);
 
-    void ComputeEigenvaluesDeformationGradient(const unsigned int& rPointNumber,
+    void ComputeEigenvaluesDeformationGradient(const unsigned int PointNumber,
                     BoundedMatrix<double,3,3>& rOrigin, BoundedMatrix<double,3,3>& rTarget, BoundedMatrix<double,3,3>& rTensor,
                     const array_1d<double, 3>& rBaseRefContraTot1, const array_1d<double, 3>& rBaseRefContraTot2,
                     const array_1d<double, 3>& rE1Tot, const array_1d<double, 3>& rE2Tot, const array_1d<double, 3>& rE3Tot,
                     const array_1d<double, 3>& rgab,
                     double& rLambda1, double& rLambda2);
 
-    void ComputeEigenvectorsDeformationGradient(const unsigned int& rPointNumber,
+    void ComputeEigenvectorsDeformationGradient(const unsigned int PointNumber,
                                 BoundedMatrix<double,3,3>& rTensor, BoundedMatrix<double,3,3>& rOrigin,
                                 const BoundedMatrix<double,3,3>& rDeformationGradientTotal,
                                 const array_1d<double, 3>& rE1Tot, const array_1d<double, 3>& rE2Tot,
                                 const double Lambda1, const double Lambda2,
                                 BoundedMatrix<double,3,3>& rNAct);
 
-    void ModifyPrestress(const unsigned int& rPointNumber,
+    void ModifyPrestress(const unsigned int PointNumber,
                     BoundedMatrix<double,3,3>& rOrigin, BoundedMatrix<double,3,3>& rTarget,BoundedMatrix<double,3,3>& rTensor,
                     const array_1d<double, 3>& rE1, const array_1d<double, 3>& rE2, const array_1d<double, 3>& rE3, const array_1d<double, 3>& rG3,
                     const array_1d<double, 3>& rg1, const array_1d<double, 3>& rg2, const array_1d<double, 3>& rg3, const BoundedMatrix<double,3,3>& rNAct,
                     const double Lambda1, const double Lambda2);
 
-    const Matrix CalculateDeformationGradient(const unsigned int& rPointNumber);
+    const Matrix CalculateDeformationGradient(const unsigned int PointNumber);
 
     int  Check(const ProcessInfo& rCurrentProcessInfo) override;
 

